app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationDataTransformer.kt

```diff
@@ -0,0 +1,353 @@
+        val preferredLanguage = AdditionalInfo(
+            name = "preferred_language",
+            stringValue = additionalInfoData.preferredLanguage?.toString()
+        )
+
+        // TODO confirming, is assistant_declaration based on "I CONFIRM THAT I HAVE... in assistant, or just "Did someone help you with this form"?
+        val assistantDeclaration = AdditionalInfo(
+            name = "assistant_declaration",
+            stringValue = if (assistanceData.hasSomeoneAssisted) {
+                if (assistanceData.hasConfirmedTerms) {
+                    "true"
+                } else {
+                    throw InvalidRegistrationException(
+                        "assistanceData.hasSomeoneAssisted is true but, did not confirm terms",
+                        R.string.confirm_terms_not_checked
+                    )
+                }
+            } else {
+                "false"
+            }
+        )
+
+        return listOf(preferredLanguage, assistantDeclaration)
+    }
+
+    private fun PersonNameData.toApiName(): Name {
+        return Name(
+            firstName = firstName,
+            lastName = lastName,
+            middleName = middleName,
+            titlePrefix = title.toString(), // TODO should this be localized?
+            titleSuffix = suffix?.toString() // TODO should this be localized?
+        )
+    }
+
+    private fun AddressData.toApiAddressData(): Address {
+        val base = NumberedThoroughfareAddress(
+            completeAddressNumber = "",
+            completeStreetName = streetAddress,
+            completeSubAddress = buildSubAddress(),
+            completePlaceNames = buildCompletePlacesNames(),
+            state = state,
+            zipCode = zipCode
+        )
+
+        return Address(base)
+    }
+
+    private fun AddressData.buildSubAddress(): List<CompleteSubAddress>? {
+        val unitType = unitType?.let {
+            if (it.isEmpty()) return@let null
+
+            val subAddressComponents = listOfNotNull(unitType, unitNumber)
+                .nullIfEmpty() ?: return@let null
+
+            CompleteSubAddress(
+                subAddressType = it,
+                // TODO Do we use the unitNumber as is, or append a transformed unitType like in the JSON example?
+                subAddress = subAddressComponents.joinToString(" ")
+            )
+        }
+
+        val line2 = streetAddressTwo?.let {
+            if (it.isEmpty()) return@let null
+            CompleteSubAddress(
+                subAddressType = "LINE2",
+                subAddress = it
+            )
+        }
+
+        return listOfNotNull(unitType, line2).nullIfEmpty()
+    }
+
+    private fun AddressData.buildCompletePlacesNames(): List<CompletePlaceName>? {
+        val city = CompletePlaceName("MunicipalJurisdiction", city)
+
+        // TODO Does county need "County" appended to match docs, eg ("Allegheny County")?
+        // TODO Is this mandatory if county is empty?
```

Comment:
I would say no. I think there's a validation that ensures county can't be empty though.